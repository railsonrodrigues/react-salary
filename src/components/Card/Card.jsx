import React from 'react';
import styled from 'styled-components';

const CardWrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  gap: 15px 0;
  width: 200px;
  height: 150px;
  border-radius: 5px;
  box-shadow: rgba(206, 25, 230, 0.2) 0px 8px 24px;
`;

const Title = styled.h3`
  font-size: 14px;
  font-weight: 400;
  color: #818181;
`;

const Amount = styled.span`
  font-size: 32px;
  font-weight: 700;
  color: #ce19e6;
`;

const Percentage = styled.span`
  font-size: 20px;
  font-weight: 400;
  color: #818181;
`;

export default function Card ({ title, value, percentage }) {
  return (
    <CardWrapper>
      <Title>{ title }</Title>
      <Amount>{ value }</Amount>
      <Percentage>{ percentage }</Percentage>
    </CardWrapper>
  )
}