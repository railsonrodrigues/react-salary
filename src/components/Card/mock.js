export default {
  title: 'INSS',
  amount: '500',
  percentage: 30,
  balance: true,
}