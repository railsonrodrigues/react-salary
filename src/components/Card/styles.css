.card {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: 200px;
  width: 200px;
  box-shadow: 0 0 10px 0 rgba(0,0,0,0.3);
}