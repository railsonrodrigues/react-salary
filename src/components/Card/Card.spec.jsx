import { fireEvent, render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import React from 'react';
import Card from './Card';

import props from './mock';

describe('<Card />', () => {
  it('should render Card correctly', () => {
    const { container } = render(<Card {...props} />);

    const cardWrapper = container.querySelector('div');
    const titleElement = container.querySelector('h3.sc-gtsrHT');
    const amountElement = container.querySelector('span.sc-dlnjwi');
    const percentageElement = container.querySelector('span.sc-hKFxyN');

    expect(cardWrapper).toBeInTheDocument();
    expect(titleElement).toBeInTheDocument();
    expect(amountElement).toBeInTheDocument();
    expect(percentageElement).toBeInTheDocument();
  })
  it('should match snapshot', () => {
    const { container } = render(<Card {...props} />);

    expect(container.firstChild).toMatchSnapshot();
  })
})
