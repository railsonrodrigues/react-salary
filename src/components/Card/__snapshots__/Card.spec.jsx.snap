// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should match snapshot 1`] = `
<div
  class="sc-bdnxRM dDYZfr"
>
  <h3
    class="sc-gtsrHT jsbkKQ"
  >
    INSS
  </h3>
  <span
    class="sc-dlnjwi gucxPP"
  />
  <span
    class="sc-hKFxyN heEyz"
  >
    30
  </span>
</div>
`;
